references/classification/train.py

```diff
@@ -261,13 +272,21 @@ def main(args):
         raise RuntimeError(f"Invalid optimizer {args.opt}. Only SGD, RMSprop and AdamW are supported.")
 
     scaler = torch.cuda.amp.GradScaler() if args.amp else None
+    
+    batches_per_epoch = len(data_loader)
+    warmup_iters = args.lr_warmup_epochs
+    total_iters = args.epochs
+    
+    if args.lr_step_every_batch:
+        warmup_iters *= batches_per_epoch
+        total_iters *= batches_per_epoch
```

Comment:
Hmmm... Though I agree that per iteration stepping gives better control, it's something that we should probably do once we migrate to a better trainer solution. The proposed change has some side-effects such as interactions with the EMA decay estimation and the overcomplication of the training loops. I think we should be able to reproduce the models without this change, even if we need to do a few more iterations to achieve it. WDYT?

Reply:
This netted me about about 0.3% improvement. I can rerun without it in the final form to get quantifiable results.

Reply:
> This netted me about about 0.3% improvement.

I think that's acceptable. You can also try to run it a little bit longer to have more gradient updates and perhaps close the gap. 

> I can rerun without it in the final form to get quantifiable results.

Sounds good!